### PR TITLE
Add some known GAT bugs as tests

### PR DESCRIPTION
**Pull request by jackh726**
In the spirit of rust-lang/compiler-team#476

These tests are marked as "check-fail", but also commented with "this should pass". This many of the open GAT issues that are accepted bugs.

r? @nikomatsakis

### CODE REVIEW

**Comment by jackh726** [comment]
Need to rebless the tests with TAITs (or maybe remove them if they aren't bugs anymore), since those outputs have changed after the lazy impl trait PR landed.

File: src/test/ui/generic-associated-types/bugs/issue-87803.stderr [hunk 1]
```diff
@@ -0,0 +1,12 @@
+error[E0195]: lifetime parameters or bounds on method `scan` do not match the trait declaration
+  --> $DIR/issue-87803.rs:20:12
```
Comment (dtolnay) [inline src/test/ui/generic-associated-types/bugs/issue-87803.stderr:2]: `fn scan` is on line 21, not line 20.

```suggestion
  --> $DIR/issue-87803.rs:21:12
```

**Review by dtolnay — CHANGES_REQUESTED** [review]
This failed in the rollup: https://github.com/rust-lang/rust/pull/93929#issuecomment-1036965155.

I saw some failures in this PR's status checks too. Not sure why those didn't get posted as a comment by @rust-log-analyzer.

```console
failures:
    [ui] ui/generic-associated-types/bugs/issue-86218.rs
    [ui] ui/generic-associated-types/bugs/issue-87803.rs
    [ui] ui/generic-associated-types/bugs/issue-89008.rs
```

**Comment by jackh726** [comment]
@dtolnay test failures from CI were caused by the lazy TAIT PR, which has been reverted

**Pull request merged**